File: src/Java/dynamic programming/118.杨辉三角.java
/*
 * @lc app=leetcode.cn id=118 lang=java
 *
 * [118] 杨辉三角
 */

// @lc code=start
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            // number list in current row
            List<Integer> level = new ArrayList<>();
            for (int j = 0; j < i + 1; j++) {
                // boarder condition
                if (j == 0 || j == i) level.add(1);
                // transfer equation (j = 0 when i = 0 so we can ignore i = 0)
                else level.add(res.get(i - 1).get(j - 1) + res.get(i - 1).get(j));
            }
            res.add(level);
        }
        return res;
    }
}
// @lc code=end

File: src/Javascript/dynamic programming/118.杨辉三角.js
/*
 * @lc app=leetcode.cn id=118 lang=javascript
 *
 * [118] 杨辉三角
 */

// @lc code=start
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function (numRows) {
    const res = [];
    for (let i = 0; i < numRows; i++) {
        // number list in current row
        const level = [];
        for (let j = 0; j < i + 1; j++) {
            // boarder condition
            if (j === 0 || j === i) level.push(1);
            // transfer equation (j = 0 when i = 0 so we can ignore i = 0)
            else level.push(res[i - 1][j - 1] + res[i - 1][j]);
        }
        res.push(level);
    }
    return res;
};
// @lc code=end

